### PR TITLE
Login features

### PR DESCRIPTION
**Pull request by brianly27**
added login, create account, logout functionality
user is rerouted if not logged in or clicks sign out button

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -6,23 +6,43 @@ import './App.css'
...
+    icon: ''
+  }
+
+  setAccountInfo = state => {
+    this.setState(state)
+  }
+
+  handleSignOut = () => {
+    this.setState({ accountId: '' })
+    window.history.back()
```
Comment (justinamaple) [inline src/App.js]:
```suggestion
    window.history.push('/login')
```

We should push them directly to the login page

File: src/App.js [hunk 2]
```diff
@@ -6,23 +6,43 @@ import './App.css'
...
   }
 
   render() {
     const { accountId, icon } = this.state
-
+    const { setAccountInfo, handleSignOut } = this
```
Comment (justinamaple) [inline src/App.js]: cool, haven't seen that before

File: src/App.test.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import App from './App'
+
+it('renders without crashing', () => {
+  const div = document.createElement('div')
+  ReactDOM.render(<App />, div)
+  ReactDOM.unmountComponentAtNode(div)
+})
```
Comment (justinamaple) [inline src/App.test.js]: Let's add some more tests :)

File: src/components/NavBar.js [hunk 4]
```diff
@@ -13,6 +13,9 @@ const NavBar = ({ handleWriteClick, handleJournalClick }) => {
         <Button onClick={handleJournalClick} className='ui button'>
           Read Journal
         </Button>
+        <Button onClick={handleSignOut} className='ui button'>
```
Comment (justinamaple) [inline src/components/NavBar.js]: This works fine for our case, but be aware that if we expand to more routes we will need to render this based on state.

File: src/containers/LoginScreen.js [hunk 5]
```diff
@@ -7,21 +7,44 @@ class LoginScreen extends Component {
...
+
   handleChange = e => {
     let change = {}
     change[e.target.name] = e.target.value
     this.setState(change)
   }
 
   createAccount = e => {
-    console.log(this.state)
+    this.fetchAccount('signup')
```
Comment (justinamaple) [inline src/containers/LoginScreen.js]: Rather then just passing the string, let's pass the entire URL

File: src/containers/LoginScreen.js [hunk 6]
```diff
@@ -7,21 +7,44 @@ class LoginScreen extends Component {
...
   loginAccount = e => {
-    console.log(this.state)
+    this.fetchAccount('login')
+  }
+
+  fetchAccount = url => {
+    const { email, password } = this.state
+    const fetchUrl = 'http://localhost:3000/' + url
+    const { setAccountInfo } = this.props
+    console.log('here')
```
Comment (justinamaple) [inline src/containers/LoginScreen.js]: Don't need these once its working :P

File: src/containers/LoginScreen.test.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import LoginScreen from 'LoginScreen'
+
+describe('login component', () => {
+  it('renders login form', () => {
+    //write test here
```
Comment (justinamaple) [inline src/containers/LoginScreen.test.js]: Add some tests!

File: src/App.js (hunk 1 above)
Comment (brianly27) [reply src/App.js]: sure, made the change

File: src/App.test.js (hunk 3 above)
Comment (brianly27) [reply src/App.test.js]: added task to project to make more tests

File: src/components/NavBar.js (hunk 4 above)
Comment (brianly27) [reply src/components/NavBar.js]: sure, we will have to make it render conditionally if we add another page.

File: src/containers/LoginScreen.js (hunk 5 above)
Comment (brianly27) [reply src/containers/LoginScreen.js]: login and create methods now pass the entire url. also removed the console.log

File: src/containers/LoginScreen.js (hunk 6 above)
Comment (brianly27) [reply src/containers/LoginScreen.js]: thanks, removed it

File: src/containers/LoginScreen.test.js (hunk 7 above)
Comment (brianly27) [reply src/containers/LoginScreen.test.js]: will do, tasks are added

**Pull request merged**